### PR TITLE
Fix preferences

### PR DESCRIPTION
**Pull request by amagdas**
Refactor preferences to be stored inside user document instead of session, this helps with returning a valid ETAG.
- [x] Fix patching preferences
- [x] Fix session based preferences removal.

### CODE REVIEW

**Comment by sivakuna-aap** [comment]
Looks like we are refactoring complete preferences. If possible can we map the URL to `users/#USER_ID#/preferences` instead of just `/preferences`?

**Comment by amagdas** [comment]
It is still tied to the session, we get prefs like /preferences/#SESSION_ID#

File: server/apps/auth/db/commands.py [hunk 1]
```diff
@@ -91,6 +91,7 @@ def run(self, username, password):
         }
         service = superdesk.get_resource_service('auth')
         id = str(service.post([credentials])[0])
+        print('Session ID:', id)
```
Comment (akintolga) [inline server/apps/auth/db/commands.py]: This looks like a left-over bit of code...

Comment (mdhaman) [inline server/apps/auth/db/commands.py]: print not needed.

Comment (amagdas) [inline server/apps/auth/db/commands.py]: Actually is needed to know the session for which the token is generated, this is  a flask command.

Comment (mdhaman) [inline server/apps/auth/db/commands.py]: :+1:

**Comment by akintolga** [comment]
:+1:

**Pull request merged**